### DISCUSSION TITLE
additional functionality for vscodium

### DISCUSSION THREAD

**Discussion by gayanper** [VSCodium/vscodium#1244] [Q&A]
Do you have any plans to add additional functionality which will not break any existing vscode APIs into vscodium source code ?
The reason i'm asking is that i find contributing to vscode take lot of time. Even though is open source project, I feels like its not totally open source like other projects like Apache Netbeans, Eclipse etc. Its more like IntelliJ community project. Thats the reason I would like to check if you have any plans to evolve this into a vscode open source fork with more communicated accepted functionality.

**Comment by daiyam** [comment]
It's a hard question for me since I've started contributed to this project for the same objective.
But I understand that VSCodium is a simple rebuilt of Visual Studio Code.

I have [my fork](https://github.com/zokugun/MrCode) which I haven't updated in a while since it's not very much used (excepted for me).
You could make your own fork or make PR on my fork (I could make time to update it)

Have you look at [Thea](https://theia-ide.org/)? It's based on the same core of Visual Studio Code. But, sadly, they don't want to touch that core done by MS (at the time of last year).

**Reply by gayanper** [reply]
I did the same, I have my own fork with feature branches and I create builds out of that my merging those changes in latest vscode master. But to make it more reachable to others I thought this repo might be a good choice if the authors of this project things that can be part of this project's vision.

**Reply by goyalyashpal** [reply]
how about creating a separate repo for it??

i think that vscodium is better the way it is - vanilla binary build of Code-OSS, and the community contrib feature is more suited to a different project build atop of VScodium

**Reply by goyalyashpal** [reply]
> sadly, they [Thea] don't want to touch that core done by MS (at the time of last year).

i don't understand this, can u rephrase?

**Reply by daiyam** [reply]
My experience: https://github.com/eclipse-theia/theia/issues/7541

**Reply by goyalyashpal** [reply]
hey! 

* i was looking at this yt video from 2021.11 [xs0haWTulrY] "Why Eclipse Theia - Marc Dumais | Eclipse Foundation" 



* and at least that video seems to imply supportion of community and mods by community

* combining both above points, it seems there's a contradiction

so:

* what does this all mean? what is theia about then?
* did you have any other interaction with theia?
* did you think about basing ur fork on theia? why or why not.

[xs0haWTulrY]: https://www.youtube.com/watch?v=xs0haWTulrY&t=13m16s "Why Eclipse Theia - Marc Dumais | Eclipse Foundation"

**Reply by daiyam** [reply]
> Theia is a framework for building IDE-like applications, while VS Code is a product so it is possible for application developers to customize their applications to suit their needs.

> In the framework we attempt to make things highly customizable and extensible meaning that any feature or behavior present can in theory be overridden/extended. In the case of the folding regions, we might rely directly on the behavior from the [monaco editor](https://microsoft.github.io/monaco-editor/) and I am unsure at the moment how one can update it easily.

I think that quote ([src](https://github.com/eclipse-theia/theia/issues/7541#issuecomment-612925972)) resume it all.
The changes that I was pushing were to deep. VSCode isn't the first or last editor to not have correct support for foldings. Hell, by default, it can't order your tabs alphabetically.
VSCode is built for the use case of Microsoft not for every devs.

Theia try to bridge that gap by allowing advanced devs to created their IDE.

VSCodium will stay as it's now (a rebuild of VSCode).

I'm thinking to use Theia for my fork since VSCode always add features/bugs that breaks my workflow...

**Reply by goyalyashpal** [reply]
> VSCodium will stay as it's now (a rebuild of VSCode).

yeah, it should stay the same way. i was not saying about modifying it. 

> I'm thinking to use Theia for my fork since VSCode always add features/bugs that breaks my workflow...

i'd be interested in seeing it unfold 😃
